File: combinatorics/resource/ST/project_5/bai_toan_4/general_graph/eal_to_al.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int n, m;
    cout << "Nhap so dinh: ";
    cin >> n;
    cout << "Nhap so canh: ";
    cin >> m;

    // Khoi tao danh sach ke
    vector<vector<int>> adjList(n);

    cout << "Nhap danh sach ke mo rong (u v w):\n";
    for (int i = 0; i < m; ++i) {
        int u, v, w;
        cin >> u >> v >> w;
        adjList[u].push_back(v); // Bo qua trong so w
    }

    cout << "Danh sach ke (AL):\n";
    for (int i = 0; i < n; ++i) {
        cout << i << ": ";
        for (int v : adjList[i])
            cout << v << " ";
        cout << "\n";
    }

    return 0;
}

File: combinatorics/resource/ST/project_5/bai_toan_4/general_graph/eal_to_al.py
n = int(input("Nhap so dinh: "))
m = int(input("Nhap so canh: "))

# Khoi tao danh sach ke
adj_list = [[] for _ in range(n)]

print("Nhap danh sach ke mo rong (u v w):")
for _ in range(m):
    u, v, w = map(int, input().split())
    adj_list[u].append(v)  # Bo trong so, chi lay v

print("Danh sach ke (AL):")
for i in range(n):
    print(f"{i}:", *adj_list[i])